lib/ansible/modules/net_tools/basics/get_url.py

```diff
@@ -182,7 +182,7 @@
   get_url:
     url: http://example.com/path/file.conf
     dest: /etc/foo.conf
-    mode: '0440'
+    mode: 0440
```

Comment:
We actually recommend quoting the mode everywhere.  Docs should be updated accordingly.

Also, there is no reason that when quoted you cannot include a leading 0.

Reply:
Got it. Yeah I know you can include the zero in quotes but that is undefined behavior (not explicit in the docs) and can lead to confusion.

If you would like I can make another example with quotes, placing it above the current example, and discourage use of the leading zero higher up.

Reply:
Quoted mode starting with a leading zero is the best example to give.  It allows the user to make one of two mistakes and the module will still do the right thing.  Only if the user makes both of those mistakes would it come out wrong.  If you feel the docs for the mode parameter should focus on quoted with leading zero, you could certainly make that PR.

Reply:
I do not think the proposed PR should be applied because of what @abadger explains.

Reply:
@abadger If you're sure that's the way forward then that's fine, though it would be too tedious for me to comb through the docs enforcing this consistency. I do think that's a redundant approach however. If you *only* need a leading zero, or *only* need quotes, why would we make it a best practice to use both at the same time? We're trading this safety buffer

Reply:
@abadger If you're sure that's the way forward then that's fine, though it would be too tedious for me to comb through the docs enforcing this consistency so I won't be undertaking such a PR. I do think that's a redundant approach however. If you only need a leading zero, or only need quotes, why would we make it a best practice to use both at the same time?

We're trading this safety buffer for additional complexity when following a consistent style based on best practices. Now the user has to make sure they get *two* things right.